ledger/ledger.go

```diff
@@ -462,11 +463,30 @@ func (l *Ledger) notifyCommit(r basics.Round) basics.Round {
 	if l.archival {
 		// Do not forget any blocks.
 		minToSave = 0
+	} else if minCatchpointsRoundsLookback := l.calcMinCatchpointRoundsLookback(); minCatchpointsRoundsLookback > 0 {
+		catchpointsMinToSave := r.SubSaturate(minCatchpointsRoundsLookback)
+		if catchpointsMinToSave < minToSave {
+			minToSave = catchpointsMinToSave
+		}
 	}
 
 	return minToSave
 }
 
+func (l *Ledger) calcMinCatchpointRoundsLookback() (minCatchpointRoundsLookback basics.Round) {
+	// cfg.StoresCatchpoints checks that CatchpointInterval is positive
+	if l.cfg.StoresCatchpoints() && l.cfg.CatchpointFileHistoryLength != 0 {
+		// Nodes catching up from the _most recent_
+		// catchpoint will need to look back at least MaxTxnLife+DeeperBlockHeaderHistory+CatchpointLookback+
+		// buffer rounds before the catchpoint round in order to build up the necessary state.
+		// The max comparison is to mitigate against small catchpoint interval configurations.
+		minCatchpointRoundsLookback = basics.Round(math.Max(float64(2*l.cfg.CatchpointInterval),
```

Comment:
I don't love casting to `float64`. I'm pretty sure that at all reasonable rounds we'll ever hit, this is fine, but I just don't want to ever have to think about floating point rounding when looking at go-algorand code.

This also probably simplifies a bit without the named return value.  Just return as soon as you have the answer. (probably invert the conditional and return 0 early.)